crates/bevy_asset/src/io/file_asset_io.rs

```diff
@@ -117,7 +117,8 @@ impl AssetIo for FileAssetIo {
         Ok(Box::new(fs::read_dir(root_path.join(path))?.map(
             move |entry| {
                 let path = entry.unwrap().path();
-                path.strip_prefix(&root_path).unwrap().to_owned()
+                pathdiff::diff_paths(path, root_path.as_path())
+                    .expect("Cannot create a relative path from 'root_path' to 'path'")
```

Comment:
```suggestion
                    .unwrap_or_else(|e| panic!("Cannot create a relative path from 'root_path' `{root_path}` to 'path' `{path}`: {e}"))
```
An explicit panic message with more context might be easier to understand the reason for a crash aused by this

Reply:
[`diff_paths`](https://docs.rs/pathdiff/latest/pathdiff/fn.diff_paths.html) returns an `Option` not a `Result`.

Reply:
> [diff_paths](https://docs.rs/pathdiff/latest/pathdiff/fn.diff_paths.html) returns an Option not a Result.

Yes, but you can still use `.expect(...)` on an `Option`. I used it just to add a little context for that potential crash.